### PR TITLE
Zgh/cupertino/fix

### PR DESCRIPTION
**Pull request by Lido233**
add some missing parts.

### CODE REVIEW

File: com.unity.uiwidgets/Runtime/cupertino/date_picker.cs [hunk 1]
```diff
@@ -1337,146 +1339,150 @@ class _CupertinoTimerPickerState : State<CupertinoTimerPicker> {
...
+                 int minute = index * widget.minuteInterval;
 
+                 string semanticsLabel = textDirectionFactor == 1
+                  ? localizations.timerPickerMinute(minute) + localizations.timerPickerMinuteLabel(minute)
+                  : localizations.timerPickerMinuteLabel(minute) + localizations.timerPickerMinute(minute);
 
-        }
-        Widget _buildMinuteColumn(EdgeInsetsDirectional additionalPadding) { 
+                return _buildPickerNumberLabel(localizations.timerPickerMinute(minute), additionalPadding);
+
```
Comment (zhuxingwei) [inline com.unity.uiwidgets/Runtime/cupertino/date_picker.cs:1475]: remove this empty line

File: com.unity.uiwidgets/Runtime/cupertino/date_picker.cs [hunk 2]
```diff
@@ -1487,45 +1493,50 @@ class _CupertinoTimerPickerState : State<CupertinoTimerPicker> {
...
+              },
+              children: CupertinoDatePickerUtils.listGenerate((int) (60 / widget.secondInterval), (int index)=> {
+                 int second = index * widget.secondInterval;
+
+                 string semanticsLabel = textDirectionFactor == 1
+                  ? localizations.timerPickerSecond(second) + localizations.timerPickerSecondLabel(second)
+                  : localizations.timerPickerSecondLabel(second) + localizations.timerPickerSecond(second);
+
+                return _buildPickerNumberLabel(localizations.timerPickerSecond(second), additionalPadding);
+
```
Comment (zhuxingwei) [inline com.unity.uiwidgets/Runtime/cupertino/date_picker.cs:1528]: remove this empty line (you can trycode cleanup with UIWidgets editor configurations)

File: com.unity.uiwidgets/Runtime/widgets/text.cs [hunk 3]
```diff
@@ -91,17 +89,15 @@ public class DefaultTextStyle : InheritedTheme {
...
             oldWidget = (DefaultTextStyle) oldWidget;
-           
             return style !=  ((DefaultTextStyle)oldWidget).style ||
-                textAlign !=  ((DefaultTextStyle)oldWidget).textAlign ||
-                softWrap !=  ((DefaultTextStyle)oldWidget).softWrap ||
-                overflow !=  ((DefaultTextStyle)oldWidget).overflow ||
-                maxLines !=  ((DefaultTextStyle)oldWidget).maxLines ||
-                textWidthBasis !=  ((DefaultTextStyle)oldWidget).textWidthBasis ||
-                textHeightBehavior !=  ((DefaultTextStyle)oldWidget).textHeightBehavior;
+                   textAlign !=  ((DefaultTextStyle)oldWidget).textAlign ||
```
Comment (zhuxingwei) [inline com.unity.uiwidgets/Runtime/widgets/text.cs:95]: we can cast oldWidget to a DefaultTextStyle widget (calling it _oldWidget) and use the latter here. e.g., textAlign != _oldWidget.textAlign.

File: engine/src/lib/ui/window/window.cc [hunk 4]
```diff
@@ -71,7 +71,7 @@ UIWIDGETS_API(char*) Window_defaultRouteName(Window* ptr) {
   const std::string routeName = ptr->client()->DefaultRouteName();
   size_t size = routeName.length() + 1;
   char* result = static_cast<char*>(malloc(size));
-  routeName.copy(result, size);
+  strcpy(result, routeName.c_str());
```
Comment (zhuxingwei) [inline engine/src/lib/ui/window/window.cc:74]: please check the potential memory leak here: the pointer result seems to be impossible to be released any more.

I am not very sure though, according to this feed https://stackoverflow.com/questions/56181049/allocate-memory-in-a-c-dll-free-in-c-sharp, we may have to deallocate the memory by hand in c++

File: com.unity.uiwidgets/Runtime/widgets/app.cs [hunk 5]
```diff
@@ -281,40 +283,7 @@ public class WidgetsApp : StatefulWidget {
         }
     }
 
-    public class WindowProvider : InheritedWidget {
```
Comment (zhuxingwei) [inline com.unity.uiwidgets/Runtime/widgets/app.cs:284]: this change is big, please check all the samples in the WidgetsSample folder, ensuring the correctness

Comment (Lido233) [reply com.unity.uiwidgets/Runtime/widgets/app.cs:284]: This class is completely deleted in the flutter 1.17.5.

File: engine/src/lib/ui/window/window.cc (hunk 4 above)
Comment (Lido233) [reply engine/src/lib/ui/window/window.cc:74]: In C# 【Window_freeDefaultRouteName(routeNamePtr); 】deallocates the memory by free() in c++.

**Pull request merged**